src/Shell/Task/ModelTask.php

```diff
@@ -253,33 +247,46 @@ public function applyAssociations($model, $associations)
     public function findBelongsTo($model, array $associations)
     {
         $schema = $model->schema();
+        $tables = $this->listAll();
+
         foreach ($schema->columns() as $fieldName) {
             if (!preg_match('/^.*_id$/', $fieldName)) {
                 continue;
             }
 
+            $tmpModelName = $this->_modelNameFromKey($fieldName);
```

Comment:
The if condition will stop the loop in case that "fieldName" is not `*_id` e.g. it will not stop it for `parent_id`

In case the core team wants this convention based composite foreign key detection I will supply test cases so that you can confirm that things work. Please let me know, soon.